Competition2017/src/competition/subsystems/shift/ShiftSubsystem.java

```diff
@@ -0,0 +1,49 @@
+package competition.subsystems.shift;
+
+import org.apache.log4j.Logger;
+
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+
+import xbot.common.command.BaseSubsystem;
+import xbot.common.controls.actuators.XSolenoid;
+import xbot.common.injection.wpi_factories.WPIFactory;
+
+@Singleton
+public class ShiftSubsystem extends BaseSubsystem {
+    private static Logger log = Logger.getLogger(ShiftSubsystem.class);
+    
+    public final XSolenoid solenoidLeft;
+    public final XSolenoid solenoidRight;
+    
+    private Gear currentGear;
+    
+    public enum Gear {
+        LOWGEAR, HIGHGEAR
+    }
+    
+    @Inject
+    public ShiftSubsystem(WPIFactory factory) {
+        log.info("Creating Shiftsubsystem");
+        
+        this.solenoidLeft = factory.getSolenoid(1);
+        this.solenoidRight = factory.getSolenoid(2);
+    }
+    
+    /**
+     * Sets the gear with the given gear
+     * @param gear a low gear or high gear that determines which gear to trigger
+     */
+    public void setGear(Gear gear) {
+        currentGear = gear;
+        if (gear == Gear.LOWGEAR) {
+            solenoidLeft.set(false);
+            solenoidRight.set(false);
+        } else if (gear == Gear.HIGHGEAR) {
+            solenoidLeft.set(true);
+            solenoidRight.set(true);
+        } else {
+            log.error(gear + " is an invalid type of gear");
```

Comment:
I'd argue that this should just be `if...else`. I'd prefer having that `else` condition because it ensures that there is a default gear state: if something goes wrong, I'd much rather have it switch to the easy-to-control low gear than do nothing.